### ISSUE TITLE
Invalid node type at position 1 for language help, sql, dot, vim

### ISSUE THREAD

**Issue by manveru** [nvim-treesitter/nvim-treesitter#3559]
### Describe the bug

When opening a file in the languages `vim`, `help`, `dot`, or `sql`, i get this error on every line.
![20220930_11h36m13s_grim](https://user-images.githubusercontent.com/3507/193241353-dfeec750-4828-42f8-bf41-79aba316327f.png)

### To Reproduce

Use neovim at commit 4dc4cf346755375e49410e16635c00a602b26c36 (though HEAD and 0.7.2 also exhibit the same behavior, I chose this version because of https://github.com/nvim-treesitter/nvim-treesitter/issues/3538#issuecomment-1259933480)
Use nvim-treesitter at commit 9bfaf62e42bdcd042df1230e9188487e62a112c0 (current HEAD)
Open any help buffer.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - WARNING: `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - WARNING: `node` executable not found (only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - OK: `git` executable found.
  - OK: `gcc` executable found. Selected from { "gcc", "cc", "gcc", "clang", "cl", "zig" }
    Version: gcc (GCC) 11.3.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - make           ✓ . ✓ . ✓
  - json           ✓ ✓ ✓ ✓ .
  - heex           ✓ ✓ ✓ ✓ ✓
  - hcl            ✓ . ✓ ✓ ✓
  - go             ✓ ✓ ✓ ✓ ✓
  - rust           ✓ ✓ ✓ ✓ ✓
  - scss           ✓ . . ✓ .
  - help           x . . . .
  - lua            ✓ ✓ ✓ ✓ ✓
  - ruby           ✓ ✓ ✓ ✓ ✓
  - graphql        ✓ . . ✓ ✓
  - cpp            ✓ ✓ ✓ ✓ ✓
  - nix            ✓ ✓ ✓ . ✓
  - regex          ✓ . . . .
  - html           ✓ ✓ ✓ ✓ ✓
  - erlang         ✓ . ✓ . .
  - markdown       ✓ . ✓ . ✓
  - bash           ✓ ✓ ✓ . ✓
  - typescript     ✓ ✓ ✓ ✓ ✓
  - php            ✓ ✓ ✓ ✓ ✓
  - sql            x . . . ✓
  - elixir         ✓ ✓ ✓ ✓ ✓
  - gomod          ✓ . . . ✓
  - http           ✓ . . . ✓
  - racket         ✓ . ✓ . ✓
  - css            ✓ . ✓ ✓ ✓
  - c              ✓ ✓ ✓ ✓ ✓
  - python         ✓ ✓ ✓ ✓ ✓
  - java           ✓ ✓ . ✓ ✓
  - embedded_template✓ . . . ✓
  - toml           ✓ ✓ ✓ ✓ ✓
  - javascript     ✓ ✓ ✓ ✓ ✓
  - zig            ✓ . ✓ ✓ ✓
  - yaml           ✓ ✓ ✓ ✓ ✓
  - dot            x . . . ✓
  - gleam          ✓ ✓ ✓ ✓ ✓
  - dockerfile     ✓ . . . ✓
  - perl           ✓ . ✓ . .
  - vim            x ✓ ✓ . ✓
  - eex            ✓ . . . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

## The following errors have been detected:
  - ERROR: help(highlights): ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 1 for language help
    help(highlights) is concatenated from the following files:
    | [ERROR]:"/nix/store/g0wsjn0xw3z5kkjqmj77ilnhpk6p7zad-vim-pack-dir/pack/myVimPackage/start/nvim-treesitter/queries/help/highlights.scm", failed to load: ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 1 for language help
  - ERROR: sql(highlights): ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 86 for language sql
    sql(highlights) is concatenated from the following files:
    | [ERROR]:"/nix/store/g0wsjn0xw3z5kkjqmj77ilnhpk6p7zad-vim-pack-dir/pack/myVimPackage/start/nvim-treesitter/queries/sql/highlights.scm", failed to load: ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 86 for language sql
  - ERROR: dot(highlights): ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 25 for language dot
    dot(highlights) is concatenated from the following files:
    | [ERROR]:"/nix/store/g0wsjn0xw3z5kkjqmj77ilnhpk6p7zad-vim-pack-dir/pack/myVimPackage/start/nvim-treesitter/queries/dot/highlights.scm", failed to load: ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 25 for language dot
  - ERROR: vim(highlights): ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 1251 for language vim
    vim(highlights) is concatenated from the following files:
    | [ERROR]:"/nix/store/g0wsjn0xw3z5kkjqmj77ilnhpk6p7zad-vim-pack-dir/pack/myVimPackage/start/nvim-treesitter/queries/vim/highlights.scm", failed to load: ...d-master/share/nvim/runtime/lua/vim/treesitter/query.lua:171: query: invalid node type at position 1251 for language vim
    |    [OK]:"/nix/store/c6fs7k9g7jy06sn24h2ilki80a8rclrm-neovim-unwrapped-master/share/nvim/runtime/queries/vim/highlights.scm"
```

### Output of `nvim --version`

```text
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by nixbld

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/nix/store/n412kw14m0gd4hqapdfzhj67vhlc06qi-neovim-unwrapped-master/share/nvim
"

Run :checkhealth for more info
```

**Comment by manveru** [comment]
Additional context, this worked fine with 9bfaf62e42bdcd042df1230e9188487e62a112c0

**Comment by goddamnparticle** [comment]
`:TSUpdate help` solved for me

**Comment by manveru** [comment]
Can confirm it works again now after updating.

**Comment by micah5** [comment]
> `:TSUpdate help` solved for me (neovim 0.8.0)

That didn't work for me, but `:TSUpdateSync` did.
